homeassistant/components/mqtt/vacuum/schema.py

```diff
@@ -15,18 +19,23 @@
 )
 
 
-def services_to_strings(services, service_to_string):
+def services_to_strings(
+    services: VacuumEntityFeature | int,
+    service_to_string: dict[VacuumEntityFeature, str],
+) -> list[str]:
     """Convert SUPPORT_* service bitmask to list of service strings."""
-    strings = []
+    strings: list[str] = []
```

Comment:
Is this annotation necessary? I'd imagine mypy is able to derive that it's a list of strings since the only thing being appended into it is the result of indexing into a dict with string values.

Reply:
`pyright` doesn't infer it because it starts empty.

The other possibility is to use a comprehension:
```python
    strings = [
        service_to_string[service]
        for service in service_to_string
        if service & services
    ]
```

Reply:
We can even leave the assignment away